### ISSUE TITLE
Use QUIC-style varints

### ISSUE THREAD

**Issue by bifurcation** [sframe-wg/sframe#48]
Instead of inventing a new variable-length integer encoding, we should consider the [scheme used in QUIC](https://tools.ietf.org/html/draft-ietf-quic-transport-17#section-16).  The size in bytes for different ranges of integers is as follows:

| Range          | SFrame | QUIC |
|:---------------|:------:|:----:|
| 0 ... 2^3 - 1  |    .5  |  1   |
|   ... 2^6 - 1  |   1.5  |  1   | 
|   ... 2^8 - 1  |   1.5  |  2   |
|   ... 2^14 - 1 |   2.5  |  2   |
|   ... 2^16 - 1 |   2.5  |  4   |
|   ... 2^24 - 1 |   3.5  |  4   |
|   ... 2^30 - 1 |   4.5  |  4   |
|   ... 2^32 - 1 |   4.5  |  8   |
|   ... 2^40 - 1 |   5.5  |  8   |
|   ... 2^48 - 1 |   6.5  |  8   |
|   ... 2^56 - 1 |   7.5  |  8   |
|   ... 2^62 - 1 |   6.5  |  8   |

Basically it doesn't pack down quite as small as the current thing, but it avoids having to deal with half-bytes.

**Comment by martinthomson** [comment]
IETF 116: discussed and dispensed with.
